src/VisualStudio/Core/Def/Implementation/FindReferences/AbstractFindUsagesCustomColumnDefinition.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.VisualStudio.Shell.TableControl;
+
+namespace Microsoft.VisualStudio.LanguageServices.FindUsages
+{
+    /// <summary>
+    /// Implementation of a custom, dynamic column for the Find All References window.
+    /// </summary>
+    internal abstract class AbstractFindUsagesCustomColumnDefinition : TableColumnDefinitionBase
+    {
+        protected AbstractFindUsagesCustomColumnDefinition()
+        {
+            DefaultColumnState = new ColumnState2(Name, isVisible: false, DefaultWidth);
+        }
+
+        public ColumnState2 DefaultColumnState { get; }
+
+        public override bool TryGetFilterItems(ITableEntryHandle entry, out IEnumerable<string> filterItems)
+        {
+            if (IsFilterable &&
+                entry.TryGetValue(Name, out var value) &&
+                value is string displayString &&
+                !string.IsNullOrEmpty(displayString))
+            {
+                filterItems = SplitColumnDisplayValue(displayString);
+                return true;
+            }
+
+            return base.TryGetFilterItems(entry, out filterItems);
+        }
```

Comment:
1. what does this method do?

Reply:
This basically determines the constituent strings for the display value in column, which should be used for applying the filter. For example, value "Read, Write" displayed in column for an entry, we will return "Read" and "Write" here, so filtering based on individual filter terms can be done.

Reply:
Added comment

---
In reply to: [220752126](https://github.com/dotnet/roslyn/pull/30155#discussion_r220752126) [](ancestors = 220752126)